Common/Algorithm/Framework/Alphas/Insight.cs

```diff
@@ -217,11 +218,17 @@ public static Guid Group(params Insight[] insights)
                 }
 
                 insight.GroupId = groupId;
-            }
 
-            return groupId;
+                yield return insight;
```

Comment:
I don't think this should be lazy. Let's remove the `yield return` syntax, otherwise users are definitely going to forget to materialize the list, for example, the existing pairs trading example doesn't materialize `Insight.Group` so I assume that is now broken by this change.